coordinator/meshapi.go

```diff
@@ -24,14 +25,17 @@ import (
 )
 
 type meshAPIServer struct {
-	grpc    *grpc.Server
-	cleanup func()
-	logger  *slog.Logger
+	grpc          *grpc.Server
+	cleanup       func()
+	getSeedEngine func() (*seedengine.SeedEngine, error)
```

Comment:
imo it's cleaner to use an interface here. I'd say the getter approach is cleaner overall.